### PR TITLE
Support doc link for the sklearn module.

### PR DESCRIPTION
**Pull request by trivialfis**
See https://scikit-learn.org/stable/developers/develop.html#developer-api-for-html-representation for more info.

It's a small enhancement for having a link to XGBoost's documentation site when an object is rendered in a jupyter notebook, see the upper right corner of the object description in the figure below.

Screenshot from 2024-08-01 23-28-19

### CODE REVIEW

**Comment by trivialfis** [comment]
hmm, we might need a newer Python for this.

**Comment by hcho3** [comment]
> hmm, we might need a newer Python for this.

How new do we need it?

File: tests/ci_build/conda_env/linux_cpu_test.yml [hunk 1]
```diff
@@ -14,7 +14,7 @@ dependencies:
 - pylint
 - numpy
 - scipy
-- scikit-learn
+- scikit-learn>=1.4.1
```
Comment (hcho3) [inline tests/ci_build/conda_env/linux_cpu_test.yml:17]: Should we also add `scikit-learn>=1.4.1` to `pyproject.toml`?

File: python-package/xgboost/sklearn.py [hunk 2]
```diff
@@ -795,6 +797,32 @@ def _more_tags(self) -> Dict[str, bool]:
...
+    def _doc_link_template(self) -> str:
+        ver = _py_version()
+        (major, minor, _), post = _parse_version(ver)
+
+        if post == "dev":
+            rel = "latest"
+        else:
+            # RTD tracks the release branch, we don't have different branch patch
+            # release.
+            rel = f"{major}.{minor}.0"
```
Comment (hcho3) [inline python-package/xgboost/sklearn.py:814]:
```suggestion
            rel = f"release_{major}.{minor}.0"
```

**Review by hcho3 — CHANGES_REQUESTED** [review]

File: tests/ci_build/conda_env/linux_cpu_test.yml (hunk 1 above)
Comment (trivialfis) [reply tests/ci_build/conda_env/linux_cpu_test.yml:17]: No need, for older versions, the code has no effect.

File: python-package/xgboost/sklearn.py (hunk 2 above)
Comment (trivialfis) [reply python-package/xgboost/sklearn.py]: Done.

**Comment by trivialfis** [comment]
Hmm, after the removal of the deterministic repartitioning script, the dask test is quite flaky. Will try to fix it in a different PR.

**Review by hcho3 — APPROVED** [review]

**Pull request merged**